### PR TITLE
Address Detekt findings

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR addresses numerous findings reported by Detekt for code in `buildSrc`.

Most of them were addressed in code. Several cases are suppressed. 

`force-jacoco` script-plugin was dropped. The [reason](https://github.com/SpineEventEngine/config/issues/214) of its existence seems to be resolved in Gradle `7.1.-rc-1`.

From [Gradle 7.1-rc-1 Release Notes](https://docs.gradle.org/7.1-rc-1/release-notes.html):

>[The JaCoCo plugin](https://docs.gradle.org/7.1-rc-1/userguide/jacoco_plugin.html) has been upgraded to the most recent [JaCoCo version 0.8.7](http://www.jacoco.org/jacoco/trunk/doc/changes.html) which includes support for Java 15 and 16 and experimental support for Java 17.

I've tried to build `base` without this plugin, and it builds fine. Our current version of Gradle fetches JaCoCo `0.8.8`. 

`Scripts.kt` referenced scripts which don't exist anymore and was also deleted.

Detekt configuration was updated as the following:

1) `UnusedPrivateMember` now passes variables named `about`.
2) `MagicNumber` was updated to include `3`. 
3) `MaxLineLength` was updated to ignore comments due to [long links issue](https://github.com/detekt/detekt/issues/5493).

### CODE REVIEW

**Comment by yevhenii-nadtochii** [comment]
@alexander-yevsyukov @armiol Please take a look.

**Comment by alexander-yevsyukov** [comment]
> UnusedPrivateMember now passes variables named about

Can we please have `ABOUT` included too (per my recent comment were you were at-mentioned)?

File: buildSrc/src/main/kotlin/force-jacoco.gradle.kts
Comment (alexander-yevsyukov) [inline buildSrc/src/main/kotlin/force-jacoco.gradle.kts:39]: Please update the `pull` script so that it removes this file when installing the latest version.

File: buildSrc/src/main/kotlin/io/spine/internal/gradle/ConfigTester.kt
Comment (alexander-yevsyukov) [inline buildSrc/src/main/kotlin/io/spine/internal/gradle/ConfigTester.kt:322]: Don't we have have it natively available from Kotlin?

File: buildSrc/src/main/kotlin/io/spine/internal/gradle/Repositories.kt
Comment (alexander-yevsyukov) [inline buildSrc/src/main/kotlin/io/spine/internal/gradle/Repositories.kt:61]: Please extract lines until `creds` into a method.

File: buildSrc/src/main/kotlin/io/spine/internal/gradle/publish/IncrementGuard.kt
Comment (alexander-yevsyukov) [inline buildSrc/src/main/kotlin/io/spine/internal/gradle/publish/IncrementGuard.kt:91]: We need more options for the main branch, just in case we'd have to default to `"main"`.

I don't want to do that because I don't like these attempts to impose "newspeak". But we may be _forced_ to change the master branch name. So, let's be ready.

File: buildSrc/src/main/kotlin/force-jacoco.gradle.kts
Comment (alexander-yevsyukov) [reply buildSrc/src/main/kotlin/force-jacoco.gradle.kts:39]: Please also take care for removal of `deps-between-tasks.kt` because it introduces conflicts with `Dependencies.kt` (to which it was renamed). But it takes manual operation to remove the file once compiler complains about it. And it's not usually obvious. So, let's same some time for our colleagues.

Comment (alexander-yevsyukov) [reply buildSrc/src/main/kotlin/force-jacoco.gradle.kts:39]: Please also don't forget to close the referenced issue.

**Review by alexander-yevsyukov — CHANGES_REQUESTED** [review]
Please see my questions and comments.

**Review by armiol — APPROVED** [review]
LGTM from my side, apart from the points raised by Alexander. 

Approving now, so that I don't block the PR even more.

File: buildSrc/src/main/kotlin/force-jacoco.gradle.kts
Comment (yevhenii-nadtochii) [reply buildSrc/src/main/kotlin/force-jacoco.gradle.kts:39]: Can we just delete the whole destination `buildSrc` and then copy it from scratch? Is this safe for all repos? In other words, do we have repos that have custom code in their `buildSrc`?

I'm asking because this clean-up looks a bit fragile:

```bash
echo "Updating Gradle 'buildSrc' scripts"
rm -f ../buildSrc/src/main/kotlin/force-jacoco.gradle.kts
rm -f ../buildSrc/src/main/kotlin/deps-between-tasks.kt
rm -f ../buildSrc/src/main/kotlin/io/spine/internal/gradle/Scripts.kt
cp -R buildSrc ..
```

Now `pull` script always copies "on top" that leads to accumulation of old files.

File: buildSrc/src/main/kotlin/io/spine/internal/gradle/ConfigTester.kt
Comment (yevhenii-nadtochii) [reply buildSrc/src/main/kotlin/io/spine/internal/gradle/ConfigTester.kt:322]: Yes, partially. Kotlin has `File.copyRecursively()` but it doesn't allow specifying exclusions. We could copy the whole directory and then delete asked `ignoredFolder` but it would come with some penalty in case `ignoredFolder` is big.

File: buildSrc/src/main/kotlin/force-jacoco.gradle.kts
Comment (alexander-yevsyukov) [reply buildSrc/src/main/kotlin/force-jacoco.gradle.kts:39]: > Is this safe for all repos? 
I don't think it's a good idea. Too aggressive. People may have custom stuff there. 

Sure, we do have Git, and our colleagues would be able to restore the deleted files. But it's too annoying.
Computers should annoy people less, not more.

Comment (alexander-yevsyukov) [reply buildSrc/src/main/kotlin/force-jacoco.gradle.kts:39]: > Now pull script always copies "on top" that leads to accumulation of old files.

That's why we need to clean up things with each next version. And then remove the cleanup code, after some time, too.

**Comment by yevhenii-nadtochii** [comment]
@alexander-yevsyukov Please take another look.

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM

**Pull request merged**
